bokehjs/src/lib/models/selections/selection.ts

```diff
@@ -90,7 +85,28 @@ export class Selection extends Model {
         this.update_through_subtraction(selection)
         break
       }
+      case "xor": {
+        this.update_through_symmetric_difference(selection)
+        break
+      }
+    }
+  }
+
+  // TODO `size` wouldn't be needed if `indices` was an instance of
+  // `Indices` class, instead of an array of numbers. Then also we
+  // could just call `.invert()` on the class.
+  invert(size: number): void {
+    const indices = new Set(this.indices)
+    const inversion = []
+    for (let i = 0; i < size; i++) {
+      if (!indices.has(i)) {
+        inversion.push(i)
+      }
     }
+    this.indices = inversion
+    // this.line_indices
+    // this.multiline_indices
+    // this.image_indices
```

Comment:
We should document somewhere that inversion does not function yet for these cases